File: pset2/ryan/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, string argv [])
{
    int q,w,e,r;
    int i = 0;
    if (argc != 2)
    {printf ("Usage: ./caesar k\n"); return 0;}
    
    int NUM = atoi(argv[1]);
    eprintf("num is %i\n",NUM);
    printf("plaintest:  ");
    string s = get_string();
    printf("ciphertext: ");
    do
    {
        if (s[i]>= 'a' && s[i] <= 'z')
        {
            if (s[i]+NUM > 'z' )
            {
            q = (s[i] + NUM) % 'z' + 'a' - 1;
            printf("%c",q);
            i++;
            }
            else
            {
            w = s[i] + NUM ;
            printf("%c",w);
            i++;
            }
        }
        else if(s[i]>= 'A' && s[i] <= 'Z')
        {
            if (s[i]+NUM > 'Z')
            {
            e = (s[i] + NUM) % 'Z' + 'A' - 1;
            printf("%c",e);
            i++;
            }
            else
            {
            r = s[i]+NUM;
            printf("%c",r);
            i++;
            }
        }
        else
        {
            printf("%c",s[i]);
            i++;
        }
    }
    while(s[i] != '\0');
printf("\n");
}

File: pset6/purification/caesar.py
import cs50
import sys


if len(sys.argv) != 2:
    print("Usage: python caesar.py k")
    exit(1)
        
k = int(sys.argv[1])
    
print("plaintext: ", end="")
p = cs50.get_string()
if p != None:
    print("ciphertext: ", end="")
    for c in p:
        if c.isalpha():
            i = ord(c)
            if c.islower():
                ch = chr((((i-97)+k)%26)+97)
                print(ch, end="")
            else:
                ch = chr((((i-65)+k)%26)+65)
                print(ch, end="")
        else:
            print(c, end="")
    print()
else:
    exit(2)
exit(0)    